spec.md

```diff
@@ -257,6 +261,12 @@ Plugins SHOULD expose all RPCs for an interface: Controller plugins SHOULD imple
 Unsupported RPCs SHOULD return an appropriate error code that indicates such (e.g. `CALL_NOT_IMPLEMENTED`).
 The full list of plugin capabilities is documented in the `ControllerGetCapabilities` and `NodeGetCapabilities` RPCs.
 
+### A Word on Quarantine States
+
+The purpose of the `QUARANTINE_S`, `QUARANTINE_P`, and `QUARANTINE_SP` states are to enable recovery from node problems.
```

Comment:
I do not see any other mention/usage of these states?

Reply:
Look at the ASCII art state transition diagrams.

Reply:
ah - I searched for word and you are using different word in ASCII diagrams. But that explanation is not enough IMO. Do you think that quarantine states should be part of `ControllerUnpublish` responses?

Reply:
Well all of the states in that ASCII chart are implicit states that the CO is supposed to follow but aren't expressed anywhere in the RPC interface. These new states are meant to be similarly implicit.

Reply:
`QUARANTINE_S`, `QUARANTINE_P`, and `QUARANTINE_SP` are in the diagrams, but I still have difficulty to understand the meeting of those states.

Reply:
All 3 states refer to situations where a particular node/volume pair is unsafe to use. The difference is which steps need to be performed to return back to the safe state. The _S state requires only unstage to return to normal, the _P state requires only unpublish to return to normal and the _SP state requires both unpublish and unstage to return to normal. These are inversions of the existing states in state diagram, with uncreatively-chosen names (I'm happy to accept better suggestions).

I wanted to emphasize that we're not merely introducing some new flags that yield different behavior, this change relies on a fundamental reordering of operations to arrive at the desired result. When a node is unreachable you have to sort out the problem by talking ONLY to the controller, and the existing CSI spec expressly forbids ControllerUnpublishing until the node side operations are complete.

Reply:
I agree with @gnufied and @bswartz . explanation of `QUARANTINE_S` , `QUARANTINE_P` , and `QUARANTINE_SP` would be better to added. I suggest adding explanations to `ControllerUnpublishVolume`  section.